docs/source/add_new_model.md

```diff
@@ -0,0 +1,573 @@
+  - What is the novel feature of the model making it different from BERT or BART (if it's an encoder-decoder model)?
+  - Which of the already existing [🤗Transformers models](https://huggingface.co/transformers/#contents) is most similar to brand_new_bert?
+  - What type of tokenizer is used? A sentence piece tokenizer? Word piece tokenizer? Is it the same tokenizer as used for BERT or BART?
+  
+ After you feel like you have gotten a good overview of the architecture of the model, you might want 
+ to write [name of hugging face teacher] for any questions you might have.
+ This might include questions regarding the model's architecture, its attention layer, etc. [name of hugging face teacher] will be more 
+ than happy to help you.
+ 
+### Next prepare your environment
+
+1. Fork the [repository](https://github.com/huggingface/transformers) by clicking on the 'Fork' button on the repository's page.
+This creates a copy of the code under your GitHub user account.
+
+2. Clone your fork to your local disk, and add the base repository as a remote:
+
+	```bash
+	git clone https://github.com/[your Github handle]/transformers.git
+	cd transformers
+	git remote add upstream https://github.com/huggingface/transformers.git
+	```
+ 
+3. Set up a development environment, for instance by running the following command:
+
+	```bash
+	conda create -n env python=3.7 --y
+	conda activate env
+	pip install -e ".[dev]"
+	```
+  
+  and return to the parent directory
+  
+  ```bash
+  cd ..
+  ```
+
+4. We recommend to add the PyTorch version of brand_new_bert to Transformers. To install PyTorch,
+  please follow the instructions on https://pytorch.org/. 
+  
+  **Note:** You don't need to have CUDA install. It is sufficient to just be working on CPU.
+
+5. To port brand_new_bert, you will also need access to its original repository:
+  
+  ```bash
+  git clone https://github.com/brand_new_bert_org/brand_new_bert.git 
+  cd brand_new_bert
+  pip install -e .
+  ```
+
+Now you have set up a development environment to port brand_new_bert to 🤗Transformers.
+  
+### Run a pretrained checkpoint using the original repository
+
+At first, you will work on the original repository. Often, the original implementation is very
+ "researchy" meaning that documentation might be lacking and the code can be difficult to understand.
+But this should be exactly your motivation to reimplement brand_new_bert. At Hugging Face, one of our main goals is to
+*make people stand on the shoulders of giants* which translates here very well into taking a working 
+model and rewriting it to make it as **accessible, user-friendly, and beautiful** as possible.
+This is the #1 motivation to re-implement models into 🤗Transformers - trying to maximize access 
+to a complex new NLP technology for **everybody**.
+	
+You should start thereby by diving into the original repository.
+
+#### Get familiar with the original repository.
```

Comment:
I am not sure where to mention this, but everything in this step can be (and should be) done on a CPU. We should clearly state that there is no need to setup a costly env with GPU before some integration training test at the very end.